posthog/models/person/util.py

```diff
@@ -148,25 +149,36 @@ def get_persons_by_uuids(team: Team, uuids: List[str]) -> QuerySet:
     return Person.objects.filter(team_id=team.pk, uuid__in=uuids)
 
 
-def delete_person(
-    person_id: UUID, properties: Dict, is_identified: bool, delete_events: bool = False, team_id: int = False
-) -> None:
+def delete_person(person: Person, delete_events: bool = False, delete_distinct_ids=False) -> None:
     timestamp = now()
 
     data = {
-        "id": person_id,
-        "team_id": team_id,
-        "properties": json.dumps(properties),
-        "is_identified": int(is_identified),
+        "id": person.uuid,
+        "team_id": person.team.id,
+        "properties": "{}",
+        "is_identified": int(person.is_identified),
         "created_at": timestamp.strftime("%Y-%m-%d %H:%M:%S"),
         "_timestamp": timestamp.strftime("%Y-%m-%d %H:%M:%S"),
+        "version": int(person.version or 0) + 100,  # keep in sync with deletePerson in plugin-server/src/utils/db/db.ts
     }
 
-    try:
-        if delete_events:
-            sync_execute(DELETE_PERSON_EVENTS_BY_ID, {"id": person_id, "team_id": team_id})
-    except:
-        pass  # cannot delete if the table is distributed
+    if delete_events:
+        sync_execute(
+            DELETE_PERSON_EVENTS_BY_ID.format(events_data_table=EVENTS_DATA_TABLE(), cluster=CLICKHOUSE_CLUSTER),
+            {"id": person.uuid, "team_id": person.team.id},
+            settings={"mutations_sync": 1 if TEST else 0},
+        )
+
+    if delete_distinct_ids:
+        distinct_id_inserts = []
+        for distinct_id in person.distinct_ids:
+            is_deleted = 1
+            version = 0
+            distinct_id_inserts.append(
+                f"('{distinct_id}', '{person.uuid}', {person.team_id}, {is_deleted}, {version}, now(), 0, 0)"
+            )
+
+        sync_execute(BULK_INSERT_PERSON_DISTINCT_ID2 + ", ".join(distinct_id_inserts), flush=False)
```

Comment:
Distinct IDs can be arbitrary – shouldn't this be interpolated?

Reply:
This is pulling from the DB where they should already have been appropriately handled _but_ nevertheless a good point, updated